src/main/java/io/quarkus/search/app/indexing/IndexingService.java

```diff
@@ -185,7 +185,14 @@ private void indexAll() {
             rollover.commit();
             Log.info("Indexing success");
         } catch (Exception e) {
-            throw new IllegalStateException("Failed to index data: " + e.getMessage(), e);
+            String message = "Failed to index data: " + e.getMessage();
+            failureCollector.critical(FailureCollector.Stage.INDEXING, message);
+            throw new IllegalStateException(message, e);
+        }
+        try {
+            failureCollector.report(indexingConfig.errorReporting());
+        } catch (IOException e) {
+            throw new IllegalStateException("Unable to report indexing failures: " + e.getMessage(), e);
         }
```

Comment:
I think this should be in a `finally` block? Though then it becomes complicated to handle cases where error reporting fails...

I'd suggest to have `FailureCollector` implement `AutoCloseabe` with `close()` doing the reporting. Then all you need to do is something like this (in `reindex`):

```java
try (FailureCollector failureCollector = new FailureCollector(indexingConfig.errorReporting())) {
    try {
            createIndexes();
            indexAll( failureCollector );
    }
    catch (RuntimeException e) {
        failureCollector.critical(e);
        throw e;
    } // close() will trigger reporting, and a failure to report will be added as suppressed exception to `e`
    finally {
        reindexingInProgress.set(false);
    }
}
```